docs/static/persistent-queues.asciidoc

```diff
@@ -42,7 +42,7 @@ When you set values for capacity and sizing settings, remember that the value yo
 TIP: If you want to define values for a specific pipeline, use <<multiple-pipelines,`pipelines.yml`>>.
 
 `queue.type`:: Specify `persisted` to enable persistent queues. By default, persistent queues are disabled (default: `queue.type: memory`).
-`path.queue`:: The directory path where the data files will be stored. By default, the files are stored in `path.data/queue`. 
+`path.queue`:: The directory path where the data files will be stored. By default, the files are stored in `path.data/queue/pipeline.id`.
```

Comment:
To me, that *is* the directory, where `path.queue/pipeline_id` is pointing to. `path.queue` (for me) is more like the *parent* folder in which each *pipeline* will have its own directory, in which all pipeline related queue files are stored.

In our setup, we also could have set `path.queue` already to individual mounted filesystems, like
```
- pipeline.id: pipeline1
  queue.max_bytes: 49392123904
  path.queue: "/opt/logstash/queue/pipeline1"
- pipeline.id: pipeline2
  queue.type: "persisted"
  queue.max_bytes: 49392123904
  path.queue: "/opt/logstash/queue/pipeline2"
- pipeline.id: pipeline3
  queue.type: "persisted"
  queue.max_bytes: 1073741824
  path.queue: "/opt/logstash/queue/pipeline3"
```

But then, the directory containing the queue files for *pipeline1* would be `/opt/logstash/queue/pipeline1/pipeline1`, as the code at https://github.com/elastic/logstash/blob/8d01b1649dc848588357d27617c12584753de32f/logstash-core/src/main/java/org/logstash/ackedqueue/QueueFactoryExt.java#L56 is creating a directory named like `pipeline_id` below of `path.queue`.

This of course also works....it's just a minimal "visual" issue.

The PR that I initially created (https://github.com/elastic/logstash/pull/14211) and @mashhurs tuned into this one, should just make sure, that the *correct* filesystem is checked and used for calculating, if enough disk capacity is available.

This PQ validator is currently breaking our setup as we have multiple pipelines running in a single logstash instance....often 2 of those pipelines have queue sizes around 45 GB assigned. The "root" filesystem (where `/opt/logstash/queue` is stored) is "only" roughly the same size (45 GB). And that is just not enough to hold all configured queue capacity of all pipelines, using the current logic.

And yes....we could also configure only a single additional filesystem holding all queue data (and that be mounted at `/opt/logstash/queue`) but experiences in the past showed that we sometimes suffered from corrupted filesystems and in the end, we lost all data on those filesystems. Therefore, we decided to use individual disks for the different pipelines. In case a filesystem "breaks", we might only loose the queue events from a single pipeline, and not all.

Setting the `path.queue` (like in example above individually for each pipeline) should also fix the validator logic. But that would mean, that we would have to move all queue data (that might still exist now for pipelines) to a "subfolder" with the same name (for every pipeline in every single logstash instance in all of our clusters) so logstash version 8 is "finding" the queue data at the right place.

As `/opt/logstash/queue/pipeline1` is a mounted filesystem and `path.queue` is set to `/opt/logstash/queue`, this is how it looks like
```
$ ls -la /opt/logstash/queue/pipeline1
total 44
drwxrwsr-x. 4 root     logstash     4096 Jun  8 06:17 .
drwxr-xr-x. 1 logstash logstash     4096 Jun  6 13:13 ..
-rw-rw-r--. 1 logstash logstash        0 Mar 16 12:45 .lock
-rw-rw-r--. 1 logstash logstash        4 Mar 16 12:45 .queue-version
-rw-r--r--. 1 logstash logstash       34 Jun  6 13:14 checkpoint.head
drwxrws---. 2 root     logstash    16384 Mar 16 12:36 lost+found
-rw-rw-r--. 1 logstash logstash 67108864 Jun  3 13:42 page.1754
```

To fix our setup, we would need to make the structure look like this
```
$ ls -la /opt/logstash/queue/pipeline1
total 44
drwxrwsr-x. 4 root     logstash     4096 Jun  8 06:17 .
drwxr-xr-x. 1 logstash logstash     4096 Jun  6 13:13 ..
drwxr-sr-x. 2 logstash logstash     4096 Jun  8 06:17 pipeline1
drwxrws---. 2 root     logstash    16384 Mar 16 12:36 lost+found
$

$ ls -la /opt/logstash/queue/pipeline1/pipeline1
total 44
drwxrwsr-x. 4 root     logstash     4096 Jun  8 06:17 .
drwxr-xr-x. 1 logstash logstash     4096 Jun  6 13:13 ..
-rw-rw-r--. 1 logstash logstash        0 Mar 16 12:45 .lock
-rw-rw-r--. 1 logstash logstash        4 Mar 16 12:45 .queue-version
-rw-r--r--. 1 logstash logstash       34 Jun  6 13:14 checkpoint.head
-rw-rw-r--. 1 logstash logstash 67108864 Jun  3 13:42 page.1754
```
This is not what we want. It would be a huge manual effort to do this.

Reply:
Thanks @rpasche for the details, I agree with your comments above that linking each FS to pipeline `path.queue` would be a workaround/solution for now.

I assume `/opt/logstash/queue` is in a different FS (not in `/dev/sdl` & `/dev/sdk`), can you plz confirm (basically `df .` should show)? I am trying to reproduce the case but my env's ability to create FSs and mounting Volumes is limited, so far no success. If you can pull my changes and run Logstash after building, would be a great help as well.

Thanks.

Reply:
> Thanks @rpasche for the details, I agree with your comments above that linking each FS to pipeline `path.queue` would be a workaround/solution for now.
> 
> I assume `/opt/logstash/queue` is in a different FS (not in `/dev/sdl` & `/dev/sdk`), can you plz confirm (basically `df .` should show)? I am trying to reproduce the case but my env's ability to create FSs and mounting Volumes is limited, so far no success. If you can pull my changes and run Logstash after building, would be a great help as well.
> 
> Thanks.

```
$ df -h

Filesystem      Size  Used Avail Use% Mounted on
overlay          60G   12G   46G  21% /
tmpfs            64M     0   64M   0% /dev
/dev/sda9        60G   12G   46G  21% /etc/hosts
shm              64M     0   64M   0% /dev/shm
tmpfs           8.0G   68K  8.0G   1% /opt/logstash/config/certs
/dev/sdl         49G   36K   49G   1% /opt/logstash/queue/pipeline1
/dev/sdk        2.0G   36K  1.9G   1% /opt/logstash/queue/pipeline2
/dev/sdj         49G   36K   49G   1% /opt/logstash/queue/pipeline3
tmpfs            32G     0   32G   0% /proc/acpi
tmpfs            32G     0   32G   0% /proc/scsi
tmpfs            32G     0   32G   0% /sys/firmware
$
```
Yes, @mashhurs , `/opt/logstash/queue` is _not_ on either **sdl**, **sdk**, **sdj**. It's on **sda** (shown here with `overlay`) which is 60 GB in size. *if* the root disk would have been big enough (for example 200GB) we wouldn't have recognized this issue.